release.py

```diff
@@ -256,11 +260,105 @@ def command_stage_docs():
 
     sys.exit(0)
 
+def command_release_announcement_email():
+    release_version_num = raw_input('What is the current release version:')
+    previos_release_version_num = raw_input('What is the previous release version:')
```

Comment:
If we run the script to do the actual release, we have this information already. It would be good to reuse this. Ie, we can keep this as-is, however add a second method that takes this information as parameters. This allow us to call the new method from here, after we collected the information, but also at the end of the regular execution of the script and pass in the information directly. Thus, if a committer does a release, it's not required to call the script again but the email template will be generated directly.

Reply:
@mjsax couple of questions
1 - "If we run the script to do the actual release, we have this information already" - 
Do you mean to say that we can get the current and previos release versions from git tags? like first fetch all the tags, sort in descending order and pick first and second indexed values are current and previos versions?

2 - "This allow us to call the new method from here, after we collected the information, but also at the end of the regular execution of the script and pass in the information directly. Thus, if a committer does a release, it's not required to call the script again but the email template will be generated directly" - 
The script 'release.py' if run, by default generates the email content for RC release for voting. At this point we are not ready for sending release completion announcement. So, what is point of generating the announcement email content at the end of the regular execution of the script 'release.py'?
I mean, we need to send release announcement email only once voting is approved and we manually tag the release branch last commit with release version.

Reply:
You definitely can determine this automatically from the existing tags. For anything with patch version > 0, it's trivial since you want the reference for previous version to be `patch_version - 1`. For the first release in a major.minor release line, you would need to figure out the correct previous major.minor.patch release and use that.

Normally I would say this is pretty easy, just list tags, find ones that match the right pattern, split segments on `.` characters, convert each to integers, and sort. However, this does get a bit messier with Kafka because of the switch in release numbering (from 4 digits in pre-1.0 to 3 digits in post-1.0), so you'd have to normalize to 4 digits, sort, then make sure you drop any extra digits from post-1.0 versions.

It'd be nice to get this all automated and the ergonomics of the script are nicer if they it is, but I wouldn't block merging this on that. This is still better than what committers do today, which is to just construct this all manually.

Reply:
Should be ok to do either 3-digit or 4-digit code (for corresponding branches) ? No need to support both in one branch IMHO